### PR TITLE
Feature/sortable tables

### PR DESCRIPTION
**Pull request by ostafinskim**
Sortable tables - initial

### CODE REVIEW

File: web/amplify/fundamentals/tables.html
Comment (NicolaSaunders) [inline web/amplify/fundamentals/tables.html:1]: @ostafinskim Please restore the all the content on this page to how it was to begin with - we will restrict all the table sorting examples and notes to the new page you have created within "Advanced Components"

Comment (ostafinskim) [reply web/amplify/fundamentals/tables.html:1]: Page restored

**Comment by NicolaSaunders** [comment]
Hi @ostafinskim - I have made some minor adjustments to the HTML code myself, which unfortunately means that I can no longer review this PR. However, I have some further comments and a request for further changes.

Please take a look at the `aria-sort` behaviour in the [APG sortable table example](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/).

* The `aria-sort` attribute should only be used on one column at a time. 
* I recommend that, before any user interaction takes place, the first column is sorted in ascending order (it currently says it is in descending order, which is incorrect). 
* No other column should have the `aria-sort` attribute until the sorting button for that column is pressed, at which point `aria-sort` should be removed from the first column header and moved to the active column

Please refer to the [notes from Adrian Roselli](https://adrianroselli.com/2021/04/sortable-table-columns.html#Col) about the use of `col` element.
* Let's use method to indicate which column is sorted, rather than adding thick coloured borders to table cells.

Please also see [Adrian's use of the `caption` element](https://adrianroselli.com/2021/04/sortable-table-columns.html#Caption) to let screen reader users know that the table can be sorted.
* Please add the words "use column header buttons to sort" inside the table caption

Please re-visit the Sass/CSS and review your use of nesting, bearing in mind our [CSS best practices](https://handbook.studio24.net/standards/our-best-practices/css-best-practices).

In terms of the JavaScript:
* I appreciate that arrow functions are the modern way of writing JS, but this is not how the rest of the JS for Amplify is written. Unless there is a specific need for them please replace the arrow functions.
* Rather than including the script for table sorting within `main.js` it would be better if it was outputted into a separate JS file. This is because it is very unlikely that this script would need to be used on every page of a website.
* Lines 13 - 17 in `main.js` should be moved into the separate JS file, as per the approach in `collapsibles.js`
* Please completely remove the existing responsive tables script. It is not as important as the table sorting scriot, and removing it will avoid any concerns about having to order the scripts in a specific way.

If you have any questions or concerns about any of the above, please let me or @sean-dunwoody know. Thanks!!

**Comment by ostafinskim** [comment]
Hello @NicolaSaunders

Thank you for valuable feedback!

I applied requested changes and they are ready to be reviewed.

If there is anything that I should be done differently please let me know and I am going to work on it once again. :)

Thanks!

**Review by NicolaSaunders — CHANGES_REQUESTED** [review]
Hi @ostafinskim 

Thanks for revisiting this. It needs some more work before I can approve it.

There may be a problem with the use of the `closest()` method, because now it is possible to change the sorting order just by clicking on the header cell - we want the function to be tied to the button inside the header cell.

Also:
* You have incorrectly removed the JS import needed for the collapsibles component and removed its function call from the `domLoadedActions()` function. Please restore both to `main.js`.
* Please remove the `responsiveTables` import from `main.js` and delete `responsive-tables.js` from the main directory
* Please remove the `timeout` variable and the associated event listener from `main.js` 
* Please move the `sortable-tables.js` file to sit outside of the main directory. It needs to be compiled into a separate JavaScript file, like `country-autocomplete.js` is. Let me know f you need help sorting this bit out, as it will require some modifications to the webpack config file.

Thanks!

**Review by NicolaSaunders — APPROVED** [review]

**Pull request merged**